types/lodash/fp.d.ts

```diff
@@ -299,7 +299,7 @@ declare namespace _ {
         <TResult>(customizer: lodash.CloneWithCustomizer<T, TResult | undefined>): TResult | T;
     }
     type LodashCloneWith2x1<T, TResult> = (value: T) => TResult | T;
-    type LodashCompact = <T>(array: lodash.List<T | null | undefined | false | "" | 0> | null | undefined) => T[];
+    type LodashCompact = <T>(array: lodash.List<T | null | undefined | false | "" | 0> | null | undefined) => Array<NonNullable<T>>;
```

Comment:
The `T | null | undefined | false | "" | 0` part of this line was trying to solve this same problem as you are (although it was written before conditional types like `NonNullable` existed). I think it would make more sense to change to something like this:

```ts
<T>(array: lodash.List<T> | null | undefined) => Array<Exclude<T, null | undefined | false | "" | 0>>;
```

That way, we aren't trying to accomplish the same thing in 2 different ways.

Reply:
Ok, I agree 👍

Reply:
`fp.d.ts` is now auto-generated and you have to modify its definition in `common` directory. I fixed this by PR#55108 but it cannot get merged due to previous errors before my PR...